src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -414,12 +424,21 @@ private function readProperty(array $zval, string $property, bool $ignoreInvalid
                         throw $e;
                     }
                 } elseif (PropertyReadInfo::TYPE_PROPERTY === $type) {
-                    if (!isset($object->$name) && !\array_key_exists($name, (array) $object)) {
+                    $valueSet = false;
+                    $bypass = $this->bypassPropertyHooks & self::BYPASS_PROPERTY_HOOK_READ && $access->hasHook() && !$access->isVirtual();
+                    $valueNotSet = !isset($object->$name) && !\array_key_exists($name, (array) $object);
```

Comment:
```suggestion
                    $initialValueNotSet = !isset($object->$name) && !\array_key_exists($name, (array) $object);
```
Otherwise, it seems to be the opposite of `$valueSet` which is not.

Reply:
Change applied